From d5af67ada0fd230a16146adc503947c0069eb159 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 12 Apr 2016 08:39:51 +1000
Subject: [PATCH 0545/1722] drm/amd/dal: drop dc_hw_sequencer abstraction

This is kinda silly, you have dce code calling a generic function
to call dce code.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/Makefile                 | 2 +-
 drivers/gpu/drm/amd/dal/dc/dce100/dce100_resource.c | 3 ++-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c | 3 ++-
 drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c | 3 ++-
 drivers/gpu/drm/amd/dal/dc/dce80/dce80_resource.c   | 4 ++--
 drivers/gpu/drm/amd/dal/dc/inc/hw_sequencer.h       | 4 ----
 6 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/Makefile b/drivers/gpu/drm/amd/dal/dc/Makefile
index a718674..e5ce905 100644
--- a/drivers/gpu/drm/amd/dal/dc/Makefile
+++ b/drivers/gpu/drm/amd/dal/dc/Makefile
@@ -26,7 +26,7 @@ AMD_DC = $(addsuffix /Makefile, $(addprefix $(FULL_AMD_DAL_PATH)/dc/,$(DC_LIBS))
 include $(AMD_DC)
 
 DISPLAY_CORE = dc.o dc_link.o dc_resource.o dc_target.o dc_sink.o dc_stream.o \
-dc_hw_sequencer.o dc_surface.o dc_link_hwss.o dc_link_dp.o dc_link_ddc.o
+dc_surface.o dc_link_hwss.o dc_link_dp.o dc_link_ddc.o
 
 AMD_DISPLAY_CORE = $(addprefix $(AMDDALPATH)/dc/core/,$(DISPLAY_CORE))
 
diff --git a/drivers/gpu/drm/amd/dal/dc/dce100/dce100_resource.c b/drivers/gpu/drm/amd/dal/dc/dce100/dce100_resource.c
index ee55e02..4149fc3 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce100/dce100_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce100/dce100_resource.c
@@ -43,6 +43,7 @@
 #include "dce110/dce110_opp.h"
 #include "dce110/dce110_clock_source.h"
 
+#include "dce100/dce100_hw_sequencer.h"
 #include "dce/dce_10_0_d.h"
 
 #ifndef mmDP_DPHY_INTERNAL_CTRL
@@ -928,7 +929,7 @@ bool dce100_construct_resource_pool(
 	}
 
 	/* Create hardware sequencer */
-	if (!dc_construct_hw_sequencer(as, dc))
+	if (!dce100_hw_sequencer_construct(dc))
 		goto stream_enc_create_fail;
 
 	return true;
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
index 27d20c9..2ca21fc 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
@@ -44,6 +44,7 @@
 #include "dce110/dce110_opp_v.h"
 #include "dce110/dce110_clock_source.h"
 
+#include "dce110/dce110_hw_sequencer.h"
 #include "dce/dce_11_0_d.h"
 
 #ifndef mmDP_DPHY_INTERNAL_CTRL
@@ -1203,7 +1204,7 @@ bool dce110_construct_resource_pool(
 
 
 	/* Create hardware sequencer */
-	if (!dc_construct_hw_sequencer(as, dc))
+	if (!dce110_hw_sequencer_construct(dc))
 		goto stream_enc_create_fail;
 
 	bw_calcs_init(&dc->bw_dceip, &dc->bw_vbios, BW_CALCS_VERSION_CARRIZO);
diff --git a/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c b/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
index 0f2bc3f..3f3ed22 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
@@ -42,6 +42,7 @@
 #include "dce110/dce110_ipp.h"
 #include "dce112/dce112_clock_source.h"
 
+#include "dce112/dce112_hw_sequencer.h"
 #include "dce/dce_11_2_d.h"
 
 #ifndef mmDP_DPHY_INTERNAL_CTRL
@@ -1242,7 +1243,7 @@ bool dce112_construct_resource_pool(
 	}
 
 	/* Create hardware sequencer */
-	if (!dc_construct_hw_sequencer(adapter_serv, dc))
+	if (!dce112_hw_sequencer_construct(dc))
 		goto stream_enc_create_fail;
 
 	bw_calcs_init(&dc->bw_dceip, &dc->bw_vbios, BW_CALCS_VERSION_BAFFIN);
diff --git a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_resource.c b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_resource.c
index 3707ea1..6b78671 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_resource.c
@@ -45,7 +45,7 @@
 #include "dce80/dce80_opp.h"
 #include "dce110/dce110_ipp.h"
 #include "dce110/dce110_clock_source.h"
-
+#include "dce80/dce80_hw_sequencer.h"
 #include "dce/dce_8_0_d.h"
 
 /* TODO remove this include */
@@ -879,7 +879,7 @@ bool dce80_construct_resource_pool(
 	}
 
 	/* Create hardware sequencer */
-	if (!dc_construct_hw_sequencer(as, dc))
+	if (!dce80_hw_sequencer_construct(dc))
 		goto stream_enc_create_fail;
 
 	return true;
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/hw_sequencer.h b/drivers/gpu/drm/amd/dal/dc/inc/hw_sequencer.h
index ba4abe0..de607df 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/hw_sequencer.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/hw_sequencer.h
@@ -132,8 +132,4 @@ struct hw_sequencer_funcs {
 		bool enable);
 };
 
-bool dc_construct_hw_sequencer(
-				struct adapter_service *adapter_serv,
-				struct core_dc *dc);
-
 #endif /* __DC_HW_SEQUENCER_H__ */
-- 
2.7.4

